File: 240523/연속 부분 합의 최댓값 구하기/max-of-partial-sum.py
n = int(input())
arr = list(map(int, input().split()))
dp = list(0 for _ in range(n))

dp[0] = arr[0]


for i in range(1, n):
    dp[i] = max(dp[i-1] + arr[i], arr[i])

print(max(dp))

File: 240627/연속 부분 합의 최댓값 구하기/max-of-partial-sum.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split("\n")

const n = parseInt(input[0])
const arr = input[1].split(" ").map(x => parseInt(x))

let dp = new Array(n).fill(0)

dp[0] = arr[0]

for (let i = 1; i < n; i++) {
    dp[i] = Math.max(arr[i], arr[i] + dp[i-1])
}

console.log(Math.max(...dp))